### DISCUSSION TITLE
Can I connect a collection to another collection while creating?

### DISCUSSION THREAD

**Discussion by samrith-s** [prisma/prisma#1913] [Q&A] [answered]
My question is, I have a resolver where I get the `troupId` in the args. Can I do something like this?

```typescript
const user = await ctx.prisma.user.create({
                data: {
                    email,
                    password: hashedPassword,
                    profile: {
                        create: {
                            firstName,
                            lastName,
                        },
                    },
                    troups: {
                        connect: {
                            id: troupId,
                        },
                    },
                    roles: {
                        create: {
                            troup: {
                                connect: {
                                    id: troupId,
                                },
                            },
                            value: 'CONTRIBUTOR',
                        },
                    },
                },
            });
```

If I run the mutation, I get this error:
```
Error:
Invalid `throw new Error()` invocation in
/Users/sam/Work/troup/server/src/resolvers/Mutations/User.mutations.ts:34:48

  30 const hashedPassword = await bcrypt.hash(password, 10);
  31
  32 const existingUser = await ctx.prisma.user.findOne({ where: { email } });
  33 if (existingUser) {
→ 34     throw new Error('User already exists!')

Error occurred during query execution:
InterpretationError("Error for binding \'3\': AssertionError(\"[Query Graph] Expected a valid parent ID to be present for a nested connect on a one-to-many relation.\")")
    at PrismaClientFetcher.request (/Users/sam/Work/troup/server/node_modules/@prisma/client/runtime/index.js:1:47359)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

**Comment by pantharshit00** [comment, marked as answer]
Hey @samrith-s 

If a record with that `troupId` exists, then you are certainly allowed to do this. 

Please open an bug report instead with a reproduction of this issue: https://github.com/prisma/prisma2/issues/new?assignees=&labels=&template=bug_report.md&title=

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
